HHS/OPRE-OPS
# Disable user accounts script

## What changed

- Adds a Python script to deactivate inactive user accounts.
- Changes are logged and user sessions, history, and event records are updated. 
- Creates a System Admin user.
- Add an ADR for the container app job and celery spikes. 
- Add a disable inactive user service to the docker compose file.
- Move audit functions used in ops db history to model file (to expose methods to the data import process).

## Issue

[#2756](https://github.com/HHS/OPRE-OPS/issues/2756)

## How to test

1. On startup, verify there is a disable-inactive-users image in the OPS container and a system user in the `ops_user` table.
2. Update a user's last_active_at timestamp to over 60 days ago.
3. Run the disable-inactive-users job.
4. Check that all user sessions have the is_active flag set to false.
5. Confirm the user's status is false and there’s a record in ops_db_history for this change.
6. Ensure a UPDATE_USER event is logged.

Verify unit and e2e tests pass. 

## Screenshots
N/A

## Definition of Done Checklist


## Links

N/A
